src/components/textInput/TextInputBase.tsx

```diff
@@ -315,12 +315,21 @@ export const TextInputBase = ({
 
   const onChangeTextHandler = useCallback(
     (text: string) => {
-      if (counterLimit && text.length > counterLimit) {
+      const actualTextLength = text.replace(/\s/g, "").length;
```

Comment:
I guess this replacement may not be always applied, what happens if the user needs to input a text with space and we're omitting from the count?